src/utils/atoms/createResourceAtom.ts

```diff
@@ -91,18 +95,22 @@ function createResourceFetcherAtom<P, T>(
 
           // cancel previous request if we have a special function for it
           if (ctx.canceller) {
-            ctx.canceller();
+            ctx.canceller(ctx);
             ctx.canceller = undefined;
           }
+          // extra dispatch allows resource subscribers to proccess cancel event and then
+          // process the response of the next event
+          if (ctx.allowCancel) dispatch(create('cancel', params));
 
           let requestAction: Action | null = null;
           try {
             let response: T;
             const fetcherResult =
               params === undefined ? fetcher() : fetcher(params);
-            if (Array.isArray(fetcherResult)) {
-              const [processor, canceller] = fetcherResult;
+            if ('processor' in fetcherResult) {
+              const { processor, canceller, allowCancel } = fetcherResult;
```

Comment:
@andrei-kucherov can you have a look?

Reply:
He approved PR, I guess I can merge it now?